EpiAware/docs/src/man/getting-started-julia.md

```diff
@@ -0,0 +1,199 @@
+
+The most common usage of environments is to create a new [_explicit_ environment](https://docs.julialang.org/en/v1/manual/code-loading/#Environments) for a project in a directory. This creates a `Project.toml` file in the directory that specifies the dependencies for the project and a `Manifest.toml` file that specifies the exact versions of the dependencies, and their underlying dependencies. We'll discuss how to set up a new environment for a project in the [REPL section](#using-the-julia-repl).
+
+Julia environments can be [stacked](https://docs.julialang.org/en/v1/manual/code-loading/#Environment-stacks). This means that you can have a _primary_ environment that is used by default, and then _secondary_ environments that are used when you activate them. This is useful when you want common packages to be available to all projects, but also want to isolate projects from each other.
+
+Most commonly the environment for a Julia installation is the primary environment, and you can add packages to this environment that you want to be available to all projects as we'll show in the [REPL section](#using-the-julia-repl).
+
+## Using the Julia REPL in projects
+
+The Julia REPL (Read-Eval-Print Loop) is an interactive programming environment that takes single user inputs (i.e., single expressions), evaluates them, and returns the result to the user.
+
+### Package management programmatically and from REPL
+
+Julia has a built-in package manager called `Pkg`, which is documented briefly [here](https://docs.julialang.org/en/v1/stdlib/Pkg/) and in more detail [here](https://pkgdocs.julialang.org/v1/). The package manager is used to install, update, and manage Julia packages and environments.
+
+You can use `Pkg` programmatically as a normal Julia package, which is often done in scripts. For example, if we wanted to install the `OrdinaryDiffEq` package as part of executing a julia script, we would add the following lines to the script:
+
+```julia
+using Pkg
+Pkg.add("OrdinaryDiffEq")
+```
+
+However, you can also use the package manager interactively from the REPL. For example, to install the `OrdinaryDiffEq` package from the REPL you can switch to package mode by typing `]` and then type `add OrdinaryDiffEq`. To exit package mode, type `backspace`.
+
+```julia-repl
+julia> ]
+(@v1.10) pkg> add OrdinaryDiffEq
+```
+
+This workflow is often more convenient than the programmatic interface, especially when setting packages you want to install to the environment for your julia installation, e.g the `@v1.10` environment for julia 1.10.
+
+By default, the environment for a julia installation is [stacked](https://docs.julialang.org/en/v1/manual/code-loading/#Environment-stacks) as a primary environment, so that the packages you install in the julia installation environment are available to all projects.
```

Comment:
Does this mean one should or shouldn't use the global environment? I've totally broken the python install on a few machines by being careless with the global environment vs a venv.

Reply:
99% _should_ use the "global" environment. In pythonic world I think its closer to using `conda active --stack myenv` (e.g. [here](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#nested-activation)).

However, I recommend have quite a sparse primary env... I can rewrite this to make it clearer.

Reply:
The reason is (iirc) https://docs.julialang.org/en/v1/manual/code-loading/#Environment-stacks

> The primary environment—i.e. the first environment in a stack—is faithfully embedded in a stacked environment. The full dependency graph of the first environment in a stack is guaranteed to be included intact in the stacked environment including the same versions of all dependencies.

And usually we won't want that in project development (IMO).

Reply:
I've rewritten this because it was misleading! Hope its clearer now